### PR TITLE
Wip/swagger

### PR DESCRIPTION
**Pull request by SebastienElet**
Do not merge, this is a work in progress.

The swagger ui is available in dev at `http://localhost:3000/api`

The todo list : 
 - [x] Handle basic auth (https://github.com/swagger-api/swagger-ui/issues/1922 and https://github.com/swagger-api/swagger-ui#custom-header-parameters---for-basic-auth-etc)
 - [ ] Handle oauth (http://swagger.io/specification/#securityDefinitionsObject)
 - [x] Handle JWT
 - [x] Uri params
 - [ ] Define query params + validator
 - [x] Port it on all endpoint
 - [ ] Dynamically build the swagger file according to the user right while getting it from the user pov /users/:popol/swagger.json, /swagger.json would only document public endpoints.
 - [x] Document payloads

### CODE REVIEW

**Comment by nfroidure** [comment]
Just realized i fucked up your nice commit conventions :/. We should automate this :).

**Comment by SebastienElet** [comment]
👍  for commit convention !

I use the [angular convention](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines) with @xavhan since i seen this talk https://www.youtube.com/watch?v=rt-9mPaYtKo

The direct benefit is to have commit split by concern (easier to cherrypick/rebase) and the final goal is to help to build a changelog, see http://keepachangelog.com/ and project like https://github.com/skywinder/github-changelog-generator

**Comment by nfroidure** [comment]
@SebastienElet just added json schema for payload and errors. We still have to check those payload in controllers for PUT endpoints and check there correctness in tests.

File: backend/public/index.html [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  <!-- Some basic translations -->
+  <!-- <script src='lang/translator.js' type='text/javascript'></script> -->
+  <!-- <script src='lang/ru.js' type='text/javascript'></script> -->
+  <!-- <script src='lang/en.js' type='text/javascript'></script> -->
+
+  <script type="text/javascript">
+    $(function () {
+      var url = window.location.search.match(/url=([^&]+)/);
+      $.getJSON('../api/v0/docs/configuration', function(config) {
```
Comment (SebastienElet) [inline backend/public/index.html]: @nfroidure this is temporary, we need to serve the swagger ui on the api prefix then we will be able to use a relative path to load the config.

**Comment by SebastienElet** [comment]
I took the idea from loopback explorer, i didn't find better/smarter way to configure the swagger-ui.

You could see loopback stuff here : https://github.com/strongloop/loopback-component-explorer/blob/4ebd792288d97bf3d3f313c293923b760494c5da/public/lib/loadSwaggerUI.js

**Comment by nfroidure** [comment]
Won't handle OAuth since we actually do not support it.

**Comment by nfroidure** [comment]
Just deployed but it doesn't work:
https://tripstory.insertafter.com/swagger/

I assume it is due to the fact the swagger-ui version is not fixed.

**Comment by nfroidure** [comment]
@SebastienElet i commented the code meanwhile https://github.com/nfroidure/TripStory/commit/5737a5a0a7b1bb037c590cea238ac9cf0dcd8fc4

**Pull request merged**